tests/Unit/Domain/CoordinateMaps/Test_ProductMapsTimeDep.cpp

```diff
@@ -0,0 +1,760 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <boost/optional.hpp>
+#include <cstddef>
+#include <memory>
+#include <pup.h>
+#include <string>
+#include <unordered_map>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/CoordinateMaps/Affine.hpp"
+#include "Domain/CoordinateMaps/ProductMapsTimeDep.hpp"
+#include "Domain/CoordinateMaps/ProductMapsTimeDep.tpp"
+#include "Domain/CoordinateMaps/Translation.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace domain {
+namespace {
+template <typename Map1, typename Map2>
+void test_product_of_2_maps_time_dep(
+    const CoordMapsTimeDependent::ProductOf2Maps<Map1, Map2>& map2d,
+    const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time,
+    const double xA, const double xB, const double xa, const double xb,
+    const double xi, const double x, const double eta, const double y,
+    const double yA, const double yB, const double ya, const double yb,
+    const std::array<double, 2>& expected_frame_velocity) noexcept {
+  const std::array<double, 2> point_A{{xA, yA}};
+  const std::array<double, 2> point_B{{xB, yB}};
+  const std::array<double, 2> point_xi{{xi, eta}};
+  const std::array<double, 2> point_a{{xa, ya}};
+  const std::array<double, 2> point_b{{xb, yb}};
+  const std::array<double, 2> point_x{{x, y}};
+
+  CHECK(map2d(point_A, time, functions_of_time) == point_a);
+  CHECK(map2d(point_B, time, functions_of_time) == point_b);
+  CHECK(map2d(point_xi, time, functions_of_time) == point_x);
+
+  CHECK(map2d.inverse(point_a, time, functions_of_time).get() == point_A);
+  CHECK(map2d.inverse(point_b, time, functions_of_time).get() == point_B);
+  CHECK_ITERABLE_APPROX(map2d.inverse(point_x, time, functions_of_time).get(),
+                        point_xi);
+
+  const double inv_jacobian_00 = (xB - xA) / (xb - xa);
+  const double inv_jacobian_11 = (yB - yA) / (yb - ya);
+  const auto inv_jac_A = map2d.inv_jacobian(point_A, time, functions_of_time);
+  const auto inv_jac_B = map2d.inv_jacobian(point_B, time, functions_of_time);
+  const auto inv_jac_xi = map2d.inv_jacobian(point_xi, time, functions_of_time);
+
+  CHECK(inv_jac_A.get(0, 0) == inv_jacobian_00);
```

Comment:
Is it obvious that these should be bitwise identical? I had thought that we shouldn't assume so when the math operations pass through different code to get the same answers.

Reply:
Since in this case the implementation should be the same given how simple the maps making up the product maps are, they should be bitwise identical I think. I could change the expected results to call the 1d maps being used in the product if you prefer.

Reply:
If you are confident that this simple case isn't going to have surprising machine-dependent floating point optimizations, I'm satisfied leaving it as is. It at least seems to do fine for the github testing machine.

Reply:
I think we do this for several of the maps, including the time-indepedent product maps and those have been okay so far, so I'm fairly confident.